web-registry/src/hooks/projects/useProjectsWithOrders.ts

```diff
@@ -0,0 +1,101 @@
+import { SellOrderInfo } from '@regen-network/api/lib/generated/regen/ecocredit/marketplace/v1/query';
+import {
+  ProjectInfo,
+  QueryProjectsResponse,
+} from '@regen-network/api/lib/generated/regen/ecocredit/v1/query';
+import shuffle from 'lodash/shuffle';
+
+import {
+  ProjectsSellOrders,
+  useProjectsSellOrders,
+} from 'pages/Projects/hooks/useProjectsSellOrders';
+import useEcocreditQuery from 'hooks/useEcocreditQuery';
+import { useQuerySellOrders } from 'hooks/useQuerySellOrders';
+
+const hasMetadata = (project: ProjectInfo): boolean => !!project.metadata;
+
+const hasSellOrders = (
+  project: ProjectInfo,
+  sellOrders: SellOrderInfo[],
+): boolean =>
+  sellOrders.some(sellOrder => sellOrder?.batchDenom?.startsWith(project.id));
+
+const prioritizeWithSellOrders = (
+  projects: ProjectInfo[],
+  sellOrders: SellOrderInfo[],
+  limit?: number,
+): ProjectInfo[] => {
+  const projectsWithOrders = projects.filter(projs =>
+    hasSellOrders(projs, sellOrders),
+  );
+  if (!limit || projectsWithOrders.length >= limit) return projectsWithOrders;
+
+  const numProjects = limit - projectsWithOrders.length;
```

Comment:
I think this variable will be a negative number most of the time, is this wanted?

Reply:
mmm... actually I think not, because it would have already returned from the function, in the previous check...

In any case, I have significantly changed the function, because I have seen a case that was not entirely correct.

Reply:
You're right, I was confused when reading the code but this will not happen indeed.